Source/WebInspectorUI/UserInterface/Views/ElementsTabContentView.js

```diff
@@ -127,6 +134,69 @@ WI.ElementsTabContentView = class ElementsTabContentView extends WI.ContentBrows
+            this.element.appendChild(this.createAccessibilityTreeButton());
+    }
+
+    createAccessibilityTreeButton()
+    {
+        let wrapper = document.createElement("span");
+        wrapper.className = "accessibility-tree-container";
+        this._accessibilityTreeToggleButton = new WI.ToggleButtonNavigationItem("accessibility-tree-button", 
+            WI.UIString("Go to Accessibility Tree"), 
+            WI.UIString("Go to DOM Tree"), 
+            "Images/Accessibility.svg", 
+            "Images/Accessibility-fill.svg"
+        );
+        this._accessibilityTreeToggleButton.imageType = WI.ButtonNavigationItem.ImageType.IMG;
+        this._accessibilityTreeToggleButton.addEventListener(WI.ButtonNavigationItem.Event.Clicked, (event) =>{
+            this._toggleTreeView();
+        }, this);
+        wrapper.appendChild(this._accessibilityTreeToggleButton.element);
+        return wrapper;
+    }
+
+    _toggleTreeView()
+    {
+        let mainFrame = WI.networkManager.mainFrame;
+        if (!mainFrame)
+            return;
+
+        let currentView = this.contentBrowser.currentContentView;
+        if (currentView?.domTreeOutline)
+            this._lastSelectedNode = currentView.domTreeOutline.selectedDOMNode();
+
+        const showingAccessibilityTree = this.contentBrowser.currentContentView?.representedObject instanceof WI.AccessibilityTree;
+        const newRepresentedObject = showingAccessibilityTree ? mainFrame.domTree : mainFrame.accessibilityTree;
+        
+        this.contentBrowser.showContentViewForRepresentedObject(newRepresentedObject);
+        this._accessibilityTreeToggleButton.toggled = !showingAccessibilityTree;
+        let newView = this.contentBrowser.currentContentView;
+        if (this._lastSelectedNode) {
+            if (newView.selectAndRevealDOMNode) {
+                newView.selectAndRevealDOMNode(this._lastSelectedNode);
+                
+                if (!showingAccessibilityTree) { 
+                    setTimeout(() => {
+                        if (newView._domTreeOutline) {
+                            newView._domTreeOutline.updateSelectionArea();
+                            
+                            let selectedTreeElement = newView._domTreeOutline.selectedTreeElement;
+                            if (selectedTreeElement && selectedTreeElement.updateSelectionArea) 
+                                selectedTreeElement.updateSelectionArea();
+                        }
+                        if (newView.layout) {
+                            newView.layout();
+                        }
+                        if (newView._domTreeOutline && newView._domTreeOutline.selectedDOMNode()) {
+                            let selectedNode = newView._domTreeOutline.selectedDOMNode();
+                            WI.domManager.hideDOMNodeHighlight();
+                            WI.domManager.highlightDOMNodeList([selectedNode]);
+                        }
+                    }, 100);
```

Comment:
A hardcoded timeout like this gives me pause (lol). Is this intended to work around some race, or is it the case that we don't stop highlighting nodes otherwise when switching to accessibility tree view?